backend/tasks/scheduled/convert_images_to_webp.py

```diff
@@ -0,0 +1,221 @@
+        Args:
+            img: PIL Image object
+        Returns:
+            Converted PIL Image object
+        """
+        if img.mode in ("RGB", "RGBA"):
+            return img
+
+        target_mode = self.MODE_CONVERSIONS.get(img.mode, "RGB")
+        return img.convert(target_mode)
+
+    def convert_to_webp(self, image_path: Path) -> bool:
+        """Convert a single image to WebP format.
+        Args:
+            image_path: Path to the source image
+        Returns:
+            True if conversion was successful, False otherwise
+        """
+        webp_path = image_path.with_suffix(".webp")
+
+        # Skip if WebP already exists
+        if webp_path.exists():
+            log.debug(f"WebP already exists for {image_path}")
+            return True
+
+        try:
+            with Image.open(image_path) as img:
+                # Convert image mode if necessary
+                img = self._convert_image_mode(img)
+
+                # Save as WebP
+                img.save(webp_path, "WEBP", quality=self.quality, optimize=True)
+                log.info(f"Created WebP version: {webp_path}")
+                return True
+
+        except Exception as exc:
+            log.error(f"Failed to create WebP version of {image_path}: {str(exc)}")
+            return False
+
+
+class ConvertImagesToWebPTask(PeriodicTask):
+    """Task to convert existing images to WebP format."""
+
+    def __init__(self):
+        super().__init__(
+            title="Convert images to WebP",
+            description="Convert existing image files (PNG, JPG, BMP, TIFF, GIF) to WebP format for better performance",
+            enabled=ENABLE_SCHEDULED_CONVERT_IMAGES_TO_WEBP,
+            manual_run=False,
+            cron_string=SCHEDULED_CONVERT_IMAGES_TO_WEBP_CRON,
+            func="tasks.scheduled.convert_images_to_webp.convert_images_to_webp_task.run",
+        )
+        self.resources_path = Path(RESOURCES_BASE_PATH)
+        self.converter = ImageConverter()
+        self._reset_counters()
+
+    def _reset_counters(self) -> None:
+        """Reset processing counters."""
+        self.processed_count = 0
+        self.error_count = 0
+        self.errors: list[str] = []
+
+    def _find_convertible_images(self) -> List[Path]:
+        """Find all convertible images in the resources directory.
+        Returns:
+            List of paths to image files that can be converted to WebP
+        """
+        if not self.resources_path.exists():
+            log.warning(f"Resources path does not exist: {self.resources_path}")
+            return []
+
+        image_files = [
+            p
+            for p in self.resources_path.rglob("**/cover/*")
+            if p.is_file() and p.suffix.lower() in ImageConverter.SUPPORTED_EXTENSIONS
+        ]
+
+        return sorted(image_files)  # Sort for consistent processing order
+
+    def _process_single_image(self, image_path: Path) -> bool:
```

Comment:
nitpick: I like Go's design where a method like this could return `(bool, str)`, where the latter is the success/error message. That way, counter updates and `errors` are changed within `run()`, reducing the need for shared state between methods.